core/common/src/MapLayerSettings.ts

```diff
@@ -292,17 +295,31 @@ export class MapLayerSettings {
       transparency: undefined !== changedProps.transparency ? changedProps.transparency : this.transparency,
       transparentBackground: undefined !== changedProps.transparentBackground ? changedProps.transparentBackground : this.transparentBackground,
       subLayers: undefined !== changedProps.subLayers ? changedProps.subLayers : this.subLayers,
-      userName: undefined !== changedProps.userName ? changedProps.userName : this.userName,
-      password: undefined !== changedProps.password ? changedProps.password : this.password,
       accessKey: undefined !== changedProps.accessKey ? changedProps.accessKey : this.accessKey,
     };
-    return MapLayerSettings.fromJSON(props)!;
+    const clone = MapLayerSettings.fromJSON(props)!;
+
+    // Clone members not part of MapLayerProps
+    clone.userName = this.userName;
+    clone.password = this.password;
+
+    return clone;
   }
 
   /** @internal */
   public displayMatches(other: MapLayerSettings): boolean {
-    if (!this.matchesNameAndUrl(other.name, other.url) || this.visible !== other.visible || this.transparency !== other.transparency || this.transparentBackground !== other.transparentBackground || this.subLayers.length !== other.subLayers.length)
+    if (!this.matchesNameAndUrl(other.name, other.url)
+      || this.visible !== other.visible
+      || this.transparency !== other.transparency
+      || this.transparentBackground !== other.transparentBackground
+      || this.subLayers.length !== other.subLayers.length) {
       return false;
+    }
+
+    if (0 !== compareStringsOrUndefined(this.userName, other.userName) ||

```

Comment:
Why not simply `if this.userName !== other.userName || this.password !== other.password)`?

Reply:
right, will update it.